### PR TITLE
chore: update CHANGELOG and set version for v1.30.1 release

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7152)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,6 +54,18 @@
 
 ## v1.31.0 - TBD
 
+## v1.31.1 - 2021-08-10
```
Comment (devjgm) [inline CHANGELOG.md:57]: I think this version should be 1.30.1.

**Review by devjgm — COMMENTED** [review]
I assume you'll need to this branch back into main after the release? Or just cherry pick a few commits to main? I'm not sure.

**Review by scotthart — COMMENTED** [review]
I'll cherry pick the cmake change commit and the changelog + version commit.

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7152#-:-MglmxGy-0qQueYhN7BO:b-jx0v4w)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @devjgm)

___
*[CHANGELOG.md, line 57 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7152#-MgllS3Sk4MOeikLrJjy-r1-57:-MglmtCr-3btuuccqJ-K:b-886gsx) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/068dab64d8fd6e0561e94b43e2a30af6b79896cc/CHANGELOG.md#L57)):*
<details><summary><i>Previously, devjgm (Greg Miller) wrote…</i></summary><blockquote>

I think this version should be 1.30.1.
</blockquote></details>

Fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -54,6 +54,18 @@
 
 ## v1.31.0 - TBD
 
+## v1.30.1 - 2021-08-10
```
Comment (devbww) [inline CHANGELOG.md:57]: For what it's worth, it doesn't look like we've ever added a "-DD" before.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7152)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @devbww and @devjgm)

___
*[CHANGELOG.md, line 57 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7152#-MgllS3Sk4MOeikLrJjy-r2-57:-Mgm6fSJAkqVT4tUhz48:b5r2l3q) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/0bbfd19756b82aae794e1c2153b295cc5ea74814/CHANGELOG.md#L57)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

For what it's worth, it doesn't look like we've ever added a "-DD" before.
</blockquote></details>

I think the extra information could be useful to some users.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**